### PR TITLE
Make `rct_windowclass` strong type `WindowClass`

### PR DESCRIPTION
**Pull request by Broxzier**
Changing it from a typedef/using to a simple enum already revealed some places where implicit conversions were done, including some where its use was nonsense (MouseInput.cpp).
The changes to the Intent class were necessary to keep things working, and this splits things up more neatly.

Changing it to a enum class revealed some more cases where enums were checked.

I think the `WindowClass` and the `WindowDetail` enum could be combined (not in this PR). There is some overlap with the values of the two enums, and it wouldn't surprise me if this caused some unintentional behaviour before.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
> I think the `WindowClass` and the `WindowDetail` enum could be combined (not in this PR). There is some overlap with the values of the two enums, and it wouldn't surprise me if this caused some unintentional behaviour before.

Pretty sure those can't be combined. I looked at that recently or maybe that was window view. Can't quite remember.

**Comment by Broxzier** [comment]
> Pretty sure those can't be combined. I looked at that recently or maybe that was window view. Can't quite remember.

Then the mixed switch statements make even less sense. At least that is somewhat cleared up by this PR then.

File: src/openrct2-ui/input/MouseInput.cpp [hunk 1]
```diff
@@ -670,7 +670,7 @@ static void InputScrollBegin(rct_window& w, rct_widgetindex widgetIndex, const S
             break;
     }
     WidgetScrollUpdateThumbs(w, widgetIndex);
-    window_invalidate_by_number(widgetIndex, w.classification);
+    window_invalidate_by_number(w.classification, w.number);
```
Comment (Gymnasiast) [inline src/openrct2-ui/input/MouseInput.cpp:673]: Nice catch!

**Review by Gymnasiast — APPROVED** [review]

File: src/openrct2-ui/input/MouseInput.cpp (hunk 1 above)
Comment (Broxzier) [reply src/openrct2-ui/input/MouseInput.cpp:673]: Yeah, this line came forward on its own now the types are stricter. 😄

**Comment by Broxzier** [comment]
I see this was squashed, but I think for refactor PRs like this where each commit does something else, it's best to use merge commits. Then, if something did break by accident, it's easier to bisect.

**Comment by tupaschoal** [comment]
Agreed!

**Pull request merged**